### PR TITLE
PS-1518 workspace files

### PR DESCRIPTION
**Pull request by pivnicek**

### CODE REVIEW

File: Reader/Reader.php [hunk 1]
```diff
@@ -87,82 +88,23 @@ public function setFormat($format)
     /**
      * @param $configuration array
      * @param $destination string Destination directory
+     * @param $storage string
      */
-    public function downloadFiles($configuration, $destination)
+    public function downloadFiles($configuration, $destination, $storage = 'local')
```
Comment (odinuv) [inline Reader/Reader.php:93]: get rid of the default value - make it required argument - which will be a bc break and major release. But that'll avoid a lot of errors

**Review by odinuv — COMMENTED** [review]
yeah i think this is a good approach

File: Reader/Reader.php (hunk 1 above)
Comment (odinuv) [reply Reader/Reader.php:93]: 👍

Comment (pivnicek) [reply Reader/Reader.php:93]: @odinuv if we're going to do that should I do the same with the `downloadTables` method which currently has the arg. `$storage = 'local'` also?

Comment (odinuv) [reply Reader/Reader.php]: feel free

**Comment by pivnicek** [comment]
alright, the storage parameter is now required for all table or file loads

File: Reader/Strategy/Files/ABSWorkspaceFilesStrategy.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                $fileInfo = $this->clientWrapper->getBasicClient()->getFile($file['id'], $fileOptions);
+                $fileDestinationPath = sprintf('%s/%s_%s', $destination, $fileInfo['id'], $fileInfo["name"]);
+                $this->logger->info(sprintf('Fetching file %s (%s).', $fileInfo['name'], $file['id']));
+                $inputs[] = [
+                    'dataFileId' => $fileInfo['id'],
+                    'destination' => $fileDestinationPath,
+                ];
+                try {
+                    // add leading slash because we write the manifests locally
+                    $this->manifestWriter->writeFileManifest($fileInfo, '/' . $fileDestinationPath . ".manifest");
```
Comment (odinuv) [inline Reader/Strategy/Files/ABSWorkspaceFilesStrategy.php:41]: why do you need a slash here? it's the same local path as in localstrategy

Comment (pivnicek) [reply Reader/Strategy/Files/ABSWorkspaceFilesStrategy.php:41]: because for ABS I needed to strip the leading slash

Comment (odinuv) [reply Reader/Strategy/Files/ABSWorkspaceFilesStrategy.php:41]: let's put it another way - what do you imagine that the workspace should look like when the files are loaded there? as i see it you're assuming, that `$destination = 'data/out/files`` and you'll have:

/data/
/data/out/
/data/out/files/
/data/out/files/my-file_12345
/data/out/files/my-file_12346
/data/out/files/my-other-file_123457

in the workspace. Am i correct?

Comment (pivnicek) [reply Reader/Strategy/Files/ABSWorkspaceFilesStrategy.php:41]: This is where the files sit in the workspace (taken from test) for example `https://testingsynapseprod.blob.core.windows.net/permanent-workspace-1802873514/tmp/docker/run-5fc547e6eaf075.58172182/download/210154596_upload/upload`

Comment (pivnicek) [reply Reader/Strategy/Files/ABSWorkspaceFilesStrategy.php:41]: So, destination path is in this case `tmp/docker/run-5fc547e6eaf075.58172182/download`

Comment (pivnicek) [reply Reader/Strategy/Files/ABSWorkspaceFilesStrategy.php:41]: not sure I fully understand :-/

**Comment by pivnicek** [comment]
@odinuv, I think it should pass now.  I'm not actually too worried that the tables are strangely split because afiak usage will mostly be of files for now.

today's changes are in the last 4 commits, hope Ii didn't forget anything

File: composer.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
     },
     "target-dir": "Keboola/InputMapping",
     "scripts": {
-        "tests": "phpunit --filter testReadFiles",
+        "tests": "phpunit",
```
Comment (pivnicek) [inline composer.json]: omfg how did that get in there, sorry :-(

**Review by odinuv — APPROVED** [review]
let's hope this is done now

**Pull request merged**